cli/command_policy_export.go

```diff
@@ -0,0 +1,127 @@
+package cli
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/snapshot/policy"
+	"github.com/pkg/errors"
+)
+
+type commandPolicyExport struct {
+	policyTargetFlags
+	filePath  string
+	overwrite bool
+
+	jsonIndent bool
+
+	stdout io.Writer
+}
+
+func (c *commandPolicyExport) setup(svc appServices, parent commandParent) {
+	cmd := parent.Command("export", "Exports the policy to the specified file, or to stdout if none is specified.")
+	cmd.Flag("to-file", "File path to export to").StringVar(&c.filePath)
+	cmd.Flag("overwrite", "Overwrite the file if it exists").BoolVar(&c.overwrite)
+
+	cmd.Flag("json-indent", "Output result in indented JSON format").Hidden().BoolVar(&c.jsonIndent)
+
+	c.policyTargetFlags.setup(cmd)
+
+	c.stdout = svc.stdout()
+
+	cmd.Action(svc.repositoryReaderAction(c.run))
+}
+
+func (c *commandPolicyExport) run(ctx context.Context, rep repo.Repository) error {
+	var output io.Writer
+	var err error
+	var deferErr error
+
+	if c.filePath != "" {
+		var file *os.File
+
+		if c.overwrite {
+			file, err = os.Create(c.filePath)
+		} else {
+			file, err = os.OpenFile(c.filePath, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0666)
+			if os.IsExist(err) {
+				return errors.Wrap(err, "file already exists and overwrite flag is not set")
+			}
+		}
+
+		if err != nil {
+			return errors.Wrap(err, "error opening file to write to")
+		}
+
+		wr := bufio.NewWriter(file)
```

Comment:
bufio is generally not needed, writing straight to `file` is fine. This avoids all the flush/sync below.